algorithm/cgal/cgal.cc

```diff
@@ -2581,45 +2312,317 @@ void print_polygon_with_holes(
+            const Point& t = segment->target();
+            if ((s != triangle.vertex(0) && s != triangle.vertex(1) &&
+                 s != triangle.vertex(2)) ||
+                (t != triangle.vertex(0) && t != triangle.vertex(1) &&
+                 t != triangle.vertex(2)) ||
+                (s != to_add.vertex(0) && s != to_add.vertex(1) &&
+                 s != to_add.vertex(2)) ||
+                (t != to_add.vertex(0) && t != to_add.vertex(1) &&
+                 t != to_add.vertex(2))) {
+              // Was not edge-to-edge.
+              return true;
+            }
+          } else if (Triangle* triangle = boost::get<Triangle>(&*result)) {
+            return true;
+          } else if (std::vector<Point>* points =
+                         boost::get<std::vector<Point>>(&*result)) {
+            return true;
+          }
+        }
+      }
+      strip = strip->parent();
+    }
+    return false;
+  }
+
+  template <typename Queue>
+  void AddA(const Polyline& polyline, Queue& queue) {
+    if (nth_a() + 1 > polyline.size()) {
+      // This prevents going further than wrapping around to the start.
+      return;
+    }
+    Point next = polyline[(nth_a() + 1) % polyline.size()];
+    if (!DoesIntersect(next, b(), a())) {
+      FT deviation = CGAL::abs(
+          (CGAL::abs(CGAL::approximate_dihedral_angle(last(), next, a(), b())) /
+           180) -
+          1);
+      FT length = CGAL::sqrt(CGAL::to_double(CGAL::squared_distance(a(), b())));
+      FT delta = 1 / (1 + deviation * length);
+      queue.push(new Strip(nth_a() + 1, nth_b(), next, b(), a(),
+                           goodness() + delta, this));
+    }
+  }
+
+  template <typename Queue>
+  void AddB(const Polyline& polyline, Queue& queue) {
+    if (nth_b() + 1 > polyline.size()) {
+      // This prevents going further than wrapping around to the start.
+      return;
+    }
+    Point next = polyline[(nth_b() + 1) % polyline.size()];
+    if (!DoesIntersect(a(), next, b())) {
+      FT deviation = CGAL::abs(
+          (CGAL::abs(CGAL::approximate_dihedral_angle(last(), next, a(), b())) /
+           180) -
+          1);
+      FT length = CGAL::sqrt(CGAL::to_double(CGAL::squared_distance(a(), b())));
+      FT delta = 1 / (1 + deviation * length);
+      queue.push(new Strip(nth_a(), nth_b() + 1, a(), next, b(),
+                           goodness() + delta, this));
+    }
+  }
+
+  size_t Release() {
+    count_ -= 1;
+    if (count_ == 0) {
+      if (parent() && parent()->Release()) {
+        delete parent_;
+        parent_ = nullptr;
+      }
+      return true;
+    }
+    return false;
   }
+
+  void Acquire() { count_ += 1; }
+
+ private:
+  const size_t nth_a_;
+  const size_t nth_b_;
+  const Point a_;
+  const Point b_;
+  const Point last_;
+  size_t count_;
+  const FT goodness_;
+  Strip* parent_;
+};
+
+class StripComparator {
+ public:
+  bool operator()(const Strip* a, const Strip* b) {
+    return a->goodness() < b->goodness();
```

Comment:
I like the function name here 🙂...it's clear, but also fun

Reply:
I was using cost, but I wanted a positive rather than negative gradient :)